### PR TITLE
media-gfx/freecad: re-add package

### PR DESCRIPTION
**Pull request by waebbl**
Pre-release version 0.19_pre with a commit date of 2020-12-31.
The current stable upstream release 0.18.5 needs qtwebkit, that's
why I didn't want to interfere with the removal of that package.

Closes: https://bugs.gentoo.org/622726
Package-Manager: Portage-3.0.13, Repoman-3.0.2
Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

add live ebuild too

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
Everything else works fine, but when I enable `oce` it won't compile.

https://dev.gentoo.org/~juippis/logs/freecad-0.19_pre20201231_build.log

File: media-gfx/freecad/freecad-0.19_pre20201231.ebuild [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+EAPI=7
+
+# vtk needs updating to use 3.9
+PYTHON_COMPAT=( python3_{7,8} )
+
+inherit check-reqs cmake desktop optfeature python-single-r1 xdg
+
+DESCRIPTION="QT based Computer Aided Design application"
+HOMEPAGE="https://www.freecadweb.org/"
```
Comment (juippis) [inline media-gfx/freecad/freecad-0.19_pre20201231.ebuild:12]: Oh also please add the Github page as 2ndary HOMEPAGE here.

**Comment by waebbl** [comment]
> Everything else works fine, but when I enable `oce` it won't compile.
> 
> https://dev.gentoo.org/~juippis/logs/freecad-0.19_pre20201231_build.log

I didn't had this problem on my tests, but I'm not sure, whether I tested building against oce only with the 0.18.5 release from my overlay.
Searching for it doesn't bring much light into it, but I noticed the oce sources don't have a TKVCAF subdir. This feature has been introduced with opencascade-7, while the current oce's compatibility is 6.9.1. The git master has already support for it, but there's no release yet.

I added the oce USE flag just a few weeks ago upon a user request, but just now found https://wiki.freecadweb.org/OpenCASCADE#Community_edition:

> However, this alternative distribution stopped active development around 2017, and lagged behind the main version in terms of features and bug fixes. For this reason, since FreeCAD v0.17, FreeCAD is compiled exclusively with OCCT, and OCE is not tested. 

While it's not true, that active development of oce has stopped, it's correct, that the development lags behind main version considerably. Because of the last sentence, I propose to remove the oce USE flag and support and depend on opencascade exclusively and will make the necessary changes.

**Review by juippis — COMMENTED** [review]
One more thing, and I'm sorry I failed to mention this last time as it was present then too, but:
```
 * Messages for package media-gfx/freecad-0.19_pre20201231:

 * QA Notice: command not found:
 * 
 * 	/bin/sh: line 1: bzr: command not found
 * 	/bin/sh: line 1: svnversion: command not found
 * 	/bin/sh: line 1: svn: command not found
 * 	/bin/sh: line 1: bzr: command not found
 * 	/bin/sh: line 1: svnversion: command not found
 * 	/bin/sh: line 1: svn: command not found
```

Could you investigate where this comes from? Seems to happen everytime regardless of `FEATURES="test"` or USE flags.

**Comment by waebbl** [comment]
> One more thing, and I'm sorry I failed to mention this last time as it was present then too, but:
> 
> ```
>  * Messages for package media-gfx/freecad-0.19_pre20201231:
> 
>  * QA Notice: command not found:
>  * 
>  * 	/bin/sh: line 1: bzr: command not found
>  * 	/bin/sh: line 1: svnversion: command not found
>  * 	/bin/sh: line 1: svn: command not found
>  * 	/bin/sh: line 1: bzr: command not found
>  * 	/bin/sh: line 1: svnversion: command not found
>  * 	/bin/sh: line 1: svn: command not found
> ```
> 
> Could you investigate where this comes from? Seems to happen everytime regardless of `FEATURES="test"` or USE flags.

@juippis Yes this happens every time. I once found the reason for this, but can't remeber just now. I'm gonna investigate the source and tell you.

As for the todo items: I remove them and put those into an external todo list.

**Comment by juippis** [comment]
(please let me know once the issue is resolved, I wouldn't want to keep building this for no reason)

**Comment by waebbl** [comment]
I'm gonna ping you

**Comment by waebbl** [comment]
@juippis The QA issue is coming from https://github.com/FreeCAD/FreeCAD/blob/0.19_pre/src/Build/CMakeLists.txt ultimately ending in https://github.com/FreeCAD/FreeCAD/blob/0.19_pre/src/Tools/SubWCRev.py#L377
If I understand it all, we should be able to change the line to sth. `vcs=[UnknownControl()]` which should remove the trials to look for any vcs tags.
I'm gonna test this and force push, if it works.

**Comment by waebbl** [comment]
@juippis ping
The package should work now, on my test, the QA issue didn't show up anymore.
I haven't backported the change to the live ebuild though. As far as I can see, there are only local git commands called (rev-list, branch, remote, log) to determine values for the version strings, and git is installed for live ebuilds, so no need for changes from my POV.

**Review by juippis — COMMENTED** [review]
All looks good to me... BUT
```
 >: freecad
FreeCAD 0.19, Libs: 0.19RUnknown
© Juergen Riegel, Werner Mayer, Yorik van Havre and others 2001-2020
FreeCAD is free and open-source software licensed under the terms of LGPL2+ license.
FreeCAD wouldn't be possible without FreeCAD community.
  #####                 ####  ###   ####  
  #                    #      # #   #   # 
  #     ##  #### ####  #     #   #  #   # 
  ####  # # #  # #  #  #     #####  #   # 
  #     #   #### ####  #    #     # #   # 
  #     #   #    #     #    #     # #   #  ##  ##  ##
  #     #   #### ####   ### #     # ####   ##  ##  ##

Could not open catalog:
 /usr/share/xerces-c/msg/XercesMessages_en_US.cat
 or /usr/share/xerces-c/msg/XercesMessages_en_US.cat
Cannot load message domain

>: equery f xerces-c | grep -i messages
/usr/msg/XercesMessages_en_US.cat

 >: mkdir -p /usr/share/xerces-c/msg/

 >: cp /usr/msg/XercesMessages_en_US.cat /usr/share/xerces-c/msg/```
And it works. 

`[ebuild   R    ] media-gfx/freecad-0.19_pre20201231::testworld  USE="addonmgr arch ccache idf image material mesh openscad pcl plot points raytracing robot ship -cloud -debug -drawing -fem -headless -inspection -part-design -path -show -spreadsheet -surface -techdraw -test* -tux" PYTHON_SINGLE_TARGET="python3_8 -python3_7"`

**Comment by waebbl** [comment]
This rather looks like an issue with xerces-c than with freecad? The `/usr/msg` location isn't FHS compliant. Also I don't have the msg directory, either in `/usr`, nor in `/usr/share/xerces-c`, I don't even have `/usr/share/xerces-c` or any `*Messages*` catalogs belonging to xerces-c.
What should I do about this?

I have those settings:
```[ebuild   R   ] dev-libs/xerces-c-3.2.3  USE="curl iconv icu threads -doc -examples -static-libs -test" CPU_FLAGS_X86="sse2"```
and
```[ebuild   R   ] media-gfx/freecad-0.19_pre20201231  USE="addonmgr arch debug drawing fem idf image inspection material mesh openscad part-design path pcl plot points raytracing robot ship show spreadsheet surface techdraw tux -ccache -cloud -headless -oce -test" PYTHON_SINGLE_TARGET="python3_8 -python3_7" ```

**Comment by juippis** [comment]
https://bugs.gentoo.org/738140 
I'll test the patch provided, and freecad after.

**Review by juippis — APPROVED** [review]

**Comment by khumarahn** [comment]
amazing, congratulations!!!

**Comment by fordfrog** [comment]
@waebbl congrats on getting freecad back into the tree!

and thank you both for this commit!

**Comment by a17r** [comment]
Next: How long will it take them to port to Qt6? :p

**Comment by waebbl** [comment]
@juippis Thank you for testing and merging!

Took a long time, but it was definitely worth the effort. Learned a lot about packaging during the last few years. Thanks to anyone who supported me in this process.

@a17r Some time after we have dropped Qt5 :-P

**Pull request closed without merging**